framework/src/modules/base_module.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+/* eslint-disable class-methods-use-this */
+
+import { Block } from '@liskhq/lisk-chain';
+import { GenesisBlock } from '@liskhq/lisk-genesis';
+import { GenesisConfig, Consensus, AccountSchema } from '../types';
+import { BaseAsset, StateStore, ReducerHandler } from './base_asset';
+
+interface Reducer {
+	[key: string]: (params: Record<string, unknown>, statestore: StateStore) => Promise<unknown>;
+}
+
+interface Action {
+	[key: string]: (params: Record<string, unknown>) => Promise<unknown>;
+}
+
+// TODO: Replace after "Update lisk-chain to support the on-chain architecture"
```

Comment:
we need a todo issue to fix major todos's 😄 
meanwhile could you add issue number here?